### PR TITLE
Feat: Deprecate payment attachment

### LINKED ISSUE #4350: [Discussion] What is an attachment
We have the following string describing our attachment shortcut:
_If this message is neither a plain text message nor a status update_

and the following message attributes are set to it:

        * :class:`telegram.Audio`
        * :class:`telegram.Dice`
        * :class:`telegram.Contact`
        * :class:`telegram.Document`
        * :class:`telegram.Animation`
        * :class:`telegram.Game`
        * :class:`telegram.Invoice`
        * :class:`telegram.Location`
        * :class:`telegram.PassportData`
        * List[:class:`telegram.PhotoSize`]
        * :class:`telegram.Poll`
        * :class:`telegram.Sticker`
        * :class:`telegram.Story`
        * :class:`telegram.SuccessfulPayment`
        * :class:`telegram.Venue`
        * :class:`telegram.Video`
        * :class:`telegram.VideoNote`
        * :class:`telegram.Voice`

Now status update is not really defined in Telegram context. From my feeling for how I would use this method, I would describe it as "user generated message which is not a plain text message". This would however mean to remove SuccessfulPayment from this list.

On the other hand, if we stick to a definition which includes SuccessfulPayment, imo all service messages which link to an object should be included.

### PR DESCRIPTION
**Pull request by Poolitzer**
closes #4350

### CODE REVIEW

File: telegram/_message.py [hunk 1]
```diff
@@ -1419,13 +1419,25 @@ def effective_attachment(
         .. versionchanged:: NEXT.VERSION
             :attr:`paid_media` is now also considered to be an attachment.
 
+        .. deprecated:: NEXT.VERSION
+            :attr:`successful_payment` will be removed in future major versions.
```
Comment (Bibo-Joshi) [inline telegram/_message.py:1423]: the "major" is a bit confusing to me. Will it still be considered in *minor* version? same below.

Comment (Poolitzer) [reply telegram/_message.py:1423]: I took that wording from https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram%2F_passport%2Fpassportfile.py#L112

My understanding is that related to the stability policy the "next" major version removes it

Comment (Bibo-Joshi) [reply telegram/_message.py:1423]: well, (git and) I will blame harshil, then :D I understand what it's supposed to mean, just stumbled on it on first read … If you like to keep it this way, I'll just merge :)

Comment (Bibo-Joshi) [reply telegram/_message.py:1423]: Generally, I prefer not to mention specific versions beforehand - just gives use more freedom in case we need it …

Comment (Poolitzer) [reply telegram/_message.py:1423]: Then Id stick with this

Comment (harshil21) [reply telegram/_message.py:1423]: yea I put "major" versions cause of the stability policy

**Pull request merged**